blockchain/message_pool/src/msgpool.rs

```diff
@@ -503,19 +499,16 @@ async fn recover_sig(
 /// in the hashmap does not yet exist, create a new mset that will correspond to the from message
 /// and push it to the pending hashmap
 async fn add_helper<T>(
-    api: Arc<RwLock<T>>,
-    bls_sig_cache: Arc<RwLock<LruCache<Cid, Signature>>>,
-    pending: Arc<RwLock<HashMap<Address, MsgSet>>>,
+    api: RwLockReadGuard<'_, T>,
+    mut bls_sig_cache: RwLockWriteGuard<'_, LruCache<Cid, Signature>>,
+    mut pending: RwLockWriteGuard<'_, HashMap<Address, MsgSet>>,
```

Comment:
sorry if misleading, I just meant you could swap the `Arc<>` with a reference, not switch the locks to the locked versions

Reply:
This is still not ideal, as you are holding all locks for this whole function when unnecessary, should just be `&RwLock<...>` and you handle the locks when necessary, as it was before you changed this, or do you have a good reason for this to remain as is?

Reply:
will change back, version I was working off of on old computer reverted it for some reason